cli/cmd/profile_test.go

```diff
@@ -30,3 +32,61 @@ func TestParseProfile(t *testing.T) {
 		t.Fatalf("ServiceProfiles are not equal: %v", err)
 	}
 }
+
+func TestValidateOptions(t *testing.T) {
+	options := newProfileOptions()
+	err := validateOptions(options)
+	exp := errors.New("You must specify exactly one of --template or --open-api")
+	if err == nil || err.Error() != exp.Error() {
+		t.Fatalf("validateOptions returned unexpected error: %s (expected: %s) for options: %+v", err, exp, options)
+	}
+
+	options = newProfileOptions()
+	err = validateOptions(options)
+	options.template = true
+	options.openAPI = "openAPI"
+	exp = errors.New("You must specify exactly one of --template or --open-api")
+	if err == nil || err.Error() != exp.Error() {
+		t.Fatalf("validateOptions returned unexpected error: %s (expected: %s) for options: %+v", err, exp, options)
+	}
+
+	options = newProfileOptions()
+	options.template = true
+	err = validateOptions(options)
+	if err != nil {
+		t.Fatalf("validateOptions returned unexpected error: %s (expected: %s) for options: %+v", err, exp, options)
+	}
+
+	options = newProfileOptions()
+	options.openAPI = "openAPI"
+	err = validateOptions(options)
+	if err != nil {
+		t.Fatalf("validateOptions returned unexpected error: %s (expected: %s) for options: %+v", err, exp, options)
+	}
+
+	options = newProfileOptions()
+	options.template = true
+	options.name = "service.name"
```

Comment:
This is an invalid service name, unfortunately:

```
$ kubectl create service clusterip service.name --clusterip="None"
The Service "service.name" is invalid: metadata.name: Invalid value: "service.name": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')
```

Maybe we can find that regex in the kubernetes codebase and use that?

Reply:
Good catch, I'll use something around `IsDNS1035Label` in the k8s codebase.